.docker/Dockerfile

```diff
@@ -0,0 +1,23 @@
+ARG PHP_VERSION=7.2
+FROM php:${PHP_VERSION}-alpine
+
+ARG XDEBUG=0
+
+COPY .docker/entrypoint.sh /usr/local/bin/
+
+RUN apk add --no-cache \
+	autoconf \
+	make \
+	g++ \
+	bash \
+	git \
+	openssl-dev
+
+RUN echo -e 'memory_limit=2G' > /usr/local/etc/php/conf.d/memory.ini
+RUN set -o pipefail && curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
```

Comment:
I think the recommended way to do this nowadays is to copy the binary from the official Docker image, see https://hub.docker.com/_/composer#php-extensions . If it slows the build down to much though, consider installing it that way: https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md#how-do-i-install-composer-programmatically-